shared/util_linux.go

```diff
@@ -330,14 +334,14 @@ func OpenPtyInDevpts(devpts_fd int, uid, gid int64) (*os.File, *os.File, error)
 
 	// Create a PTS pair.
 	if devpts_fd >= 0 {
-		fd, err := unix.Openat(devpts_fd, "ptmx", os.O_RDWR|unix.O_CLOEXEC, 0)
+		fd, err := unix.Openat(devpts_fd, "ptmx", os.O_RDWR|unix.O_CLOEXEC|NOCTTY, 0)
 		if err == nil {
 			ptx = os.NewFile(uintptr(fd), "/dev/pts/ptmx")
 		}
 	} else {
-		ptx, err = os.OpenFile("/dev/ptmx", os.O_RDWR|unix.O_CLOEXEC, 0)
-		if err != nil {
-			return nil, nil, err
+		fd, err := unix.Openat(-1, "/dev/ptmx", os.O_RDWR|unix.O_CLOEXEC|NOCTTY, 0)
+		if err == nil {
+			ptx = os.NewFile(uintptr(fd), "/dev/pts/ptmx")
 		}
```

Comment:
Aren't we missing the error handling then? What happens if the Openat fails?

Reply:
Fixed this up and rewrote parts.